### ISSUE TITLE
[BUG]: Error: The terraform-provider-github_v5.32.0 plugin crashed!

### ISSUE THREAD

**Issue by NicholasFiorentini** [integrations/terraform-provider-github#1821]
### Describe the need

I use a Terraform Cloud pipeline to configure some GitHub repositories for my organization. This pipeline was running fine (last successful run was 3 weeks ago). Today I added a new configuration, reusing the same resources, but the Terrafom Cloud plan failed with a crash error (see log):

```
Stack trace from the terraform-provider-github_v5.32.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x1017a30]
```

Environment:
```
Terraform 1.4.6
Running on Terraform Cloud
```

`provider.tf`:
```hcl
terraform {
  backend "remote" {
    organization = "xxx-bio"
    workspaces {
      name = "github-configs"
    }
  }
  required_providers {
    github = {
      source  = "integrations/github"
      version = "~> 5.0"
    }
  }
  required_version = ">= 1.4.0"
}

provider "github" {
  owner = "xxx"
}
```

### SDK Version

terraform-provider-github_v5.32.0 

### Relevant log output

```shell
Terraform v1.4.6
on linux_amd64
Initializing plugins and modules...
data.terraform_remote_state.xxx_workspaces_test: Reading...
data.terraform_remote_state.xxx_workspaces_prod: Reading...
module.infrastructure_repo_github_env["xxx-containers-test"].github_repository_environment.repo_environment: Refreshing state... [id=xxx-containers:test]
module.infrastructure_repo_github_env["xxx-containers-prod"].github_repository_environment.repo_environment: Refreshing state... [id=xxx-containers:prod]
github_actions_secret.xxx_secret["xxx"]: Refreshing state... [id=xxx:xxx]
github_actions_secret.xxx_secret["xxx.ds.preprocess"]: Refreshing state... [id=xxx.ds.preprocess:xxx]
data.terraform_remote_state.xxx_workspaces_test: Read complete after 0s
data.terraform_remote_state.xxx_workspaces_prod: Read complete after 0s

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Plugin did not respond
│ 
│   with github_actions_secret.xxx_secret["xxx.ds.preprocess"],
│   on submodules_cicd.tf line 5, in resource "github_actions_secret" "xxx_secret":
│    5: resource "github_actions_secret" "xxx_secret" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with github_actions_secret.xxx_secret["xxx"],
│   on submodules_cicd.tf line 5, in resource "github_actions_secret" "xxx_secret":
│    5: resource "github_actions_secret" "xxx_secret" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

...

╷
│ Error: Plugin did not respond
│ 
│   with module.infrastructure_repo_github_env["xxx-containers-test"].github_repository_environment.repo_environment,
│   on github_repo_environment/main.tf line 1, in resource "github_repository_environment" "repo_environment":
│    1: resource "github_repository_environment" "repo_environment" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-github_v5.32.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x1017a30]

goroutine 41 [running]:
github.com/integrations/terraform-provider-github/v5/github.resourceGithubRepositoryEnvironmentRead(0xc00063c380, {0x10c8000?, 0xc000730440?})
	github.com/integrations/terraform-provider-github/v5/github/resource_github_repository_environment.go:137 +0x4b0
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0002a63c0, 0xc00063a050, {0x10c8000, 0xc000730440})
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:470 +0x1aa
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc000293470, {0xc000aabe00?, 0x4b87a6?}, 0xc000aabe00)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:535 +0x34b
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler({0x12eeda0?, 0xc000293470}, {0x16abbf8, 0xc000ad5110}, 0xc000ad2690, 0x0)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3269 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00052c1e0, {0x16afe38, 0xc000612340}, 0xc00035a120, 0xc0005888d0, 0x1f57250, 0x0)
	google.golang.org/grpc@v1.55.0/server.go:1337 +0xde3
google.golang.org/grpc.(*Server).handleStream(0xc00052c1e0, {0x16afe38, 0xc000612340}, 0xc00035a120, 0x0)
	google.golang.org/grpc@v1.55.0/server.go:1714 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.55.0/server.go:959 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.55.0/server.go:957 +0x18c

Error: The terraform-provider-github_v5.32.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

Operation failed: failed running terraform plan (exit 1)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by NicholasFiorentini** [comment]
Possible duplicate of https://github.com/integrations/terraform-provider-github/issues/1818?

**Comment by NicholasFiorentini** [comment]
This was working 23 days ago.

**Comment by phyzical** [comment]
we just started running into this too a few days ago but we were pinned to version 5.25.1 and ci just began failing, and ours is around `resourceGithubRepositoryRead` instead of `resourceGithubRepositoryEnvironmentRead`

one thing we did notice is for us its started around the time that 1.5.5 was released, but rolling back to 1.5.4 doesn't seem to make a difference

We also did try the latest version 5.32.0 but got the same issue

Terraform 1.5.5

```

Stack trace from the terraform-provider-github_v5.25.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104ac63cc]

goroutine 304 [running]:
github.com/integrations/terraform-provider-github/v5/github.resourceGithubRepositoryRead(0x140008667e0, {0x104ed5620?, 0x1400063a400?})
        github.com/integrations/terraform-provider-github/v5/github/resource_github_repository.go:664 +0xe6c
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0x14000020640, 0x1400015cb40, {0x104ed5620, 0x1400063a400})
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:470 +0x1ac
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0x14000306490, {0x14000190c00?, 0x0?}, 0x14000190c00)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:535 +0x338
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler({0x1050d0be0?, 0x14000306490}, {0x10512d070, 0x14000b16f60}, 0x14000866380, 0x0)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3269 +0x174
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002d2000, {0x105130ee8, 0x140006424e0}, 0x14000b24fc0, 0x140003ed320, 0x10592d6b0, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xb90
google.golang.org/grpc.(*Server).handleStream(0x140002d2000, {0x105130ee8, 0x140006424e0}, 0x14000b24fc0, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x298

Error: The terraform-provider-github_v5.25.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

make: *** [plan] Error 1
```

**Comment by ivan-kiselev** [comment]
Same, can't downgrade the version since we want to use `github_repository_environment_deployment_policy` and it only is present in the latest two versions, both of them are are crashing int he described way

**Comment by NicholasFiorentini** [comment]
Update: it seems an expired GitHub PAT causes this. We generated a new token, and the provider works. Maybe the provider should not crash when the token is expired.

**Comment by muawiakh** [comment]
Fixed in `5.40.0`: https://github.com/integrations/terraform-provider-github/pull/1932

Provider returns:

```
403 Resource not accessible by integration []
```
